app/services/course_creation_step_service.rb

```diff
@@ -15,7 +15,12 @@ def get_next_step(steps, current_step)
 
   def get_previous_step(steps, current_step)
     previous_step_index = steps.find_index(current_step).pred
-    steps[previous_step_index]
+
+    if previous_step_index == -1
```

Comment:
I think it's more readable if we add this as a discrete step to each workflow rather than making it the exception to the rule

Reply:
:thinking: yeah that's true, i went back and forth. Sounds good :+1: